### DISCUSSION TITLE
Some things of interest when joining lobbies.

### DISCUSSION THREAD

**Discussion by Totoqoe** [winstxnhdw/lc-hax#468] [General]
Remember how I can use a network emulator to throttle my computers network? I've done a bit more experiments with it and found.

1. You can delay loading into the lobby until after the match starts, essentially guaranteeing that you would be the last player to join the lobby for `antikick` shenanigans.  It also has the added bonus of glitching the ship cameras which adds a bit more invisibility to your presence.
2. From what I have experienced, you're completely invisibile/unkickable until you fully load into the lobby and escape the void screen.  Even the new anti cheat mod which has an `antikick` detection can't kick you out until you fully load in. 

Speaking of the new anti cheat, it seems that it detects empty parameters when someone joins. 
![Screenshot_20240313_153142](https://github.com/winstxnhdw/lc-hax/assets/158114082/ff447ee0-9377-4c11-8d25-1be9831bf487)

Would it be possible to have `SendNewPlayerValues` send dummy data or to delay sending a response to `SendNewPlayerValues` for an indefinite amount of time if possible? Sorry if this sounds really dumb.

**Comment by winstxnhdw** [comment]
Yeah, just add a coroutine in there and add a delay.

**Reply by winstxnhdw** [reply]
Honestly, if this works, it'd be an amazing find.

**Reply by Totoqoe** [reply]
:O I hope I didn't do a dumb

**Reply by winstxnhdw** [reply]
You are one of the few people who has never done a dumb here 👍🏼

**Comment by joep26020** [comment]
this is sick! Winston can u add this I really wanna test it out

**Reply by winstxnhdw** [reply]
This weekend.

**Reply by joep26020** [reply]
im too impatient for that lol

**Reply by winstxnhdw** [reply]
You can try Toto’s network throttler in the meantime.

**Comment by joep26020** [comment]
> Remember how I can use a network emulator to throttle my computers network? I've done a bit more experiments with it and found.
> 
> 1. You can delay loading into the lobby until after the match starts, essentially guaranteeing that you would be the last player to join the lobby for `antikick` shenanigans.  It also has the added bonus of glitching the ship cameras which adds a bit more invisibility to your presence.
> 2. From what I have experienced, you're completely invisibile/unkickable until you fully load into the lobby and escape the void screen.  Even the new anti cheat mod which has an `antikick` detection can't kick you out until you fully load in.
> 
> Speaking of the new anti cheat, it seems that it detects empty parameters when someone joins. ![Screenshot_20240313_153142](https://private-user-images.githubusercontent.com/158114082/312613007-ff447ee0-9377-4c11-8d25-1be9831bf487.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAzNjM5NjYsIm5iZiI6MTcxMDM2MzY2NiwicGF0aCI6Ii8xNTgxMTQwODIvMzEyNjEzMDA3LWZmNDQ3ZWUwLTkzNzctNGMxMS04ZDI1LTFiZTk4MzFiZjQ4Ny5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMxM1QyMTAxMDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03ZjgyNGE5YjhjZDQ5OGFjODdhNjM2YWQ1ODRlYjA5YTVhYjI2MWVlM2MyOWRjNGMxZmMxZWU5NTc2YTcxNzQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.nqyUDwt5nBm2r5i9QMOabT4pjcAwWPwdxFwpOLqhQoI)
> 
> Would it be possible to have `SendNewPlayerValues` send dummy data or to delay sending a response to `SendNewPlayerValues` for an indefinite amount of time if possible? Sorry if this sounds really dumb.

Toto could you teach me how to do this?

**Reply by Totoqoe** [reply]
I would probably recommend waiting since this program throttles all of your network, the only thing that could be useful using it is delaying your join until the lobby starts, otherwise, you wont be able to do anything since you would be stuck in the wall, plus if you wait too long, you'll lose connection to steam services which kicks you out of a lobby, itll be better once it's implemented into lc hax where this issue would be cleared presumably.  

If ya still wanna give it a try, the program is called SoftPerfect Connection Simulator, it has a free demo that let's you throttle your network 30 seconds at a time, otherwise you would have to pay like $100 for the full version or pirate it like I did (wouldn't recommend since all of the keygens are so sketchy). Let me know if ya get it so I can share my settings.

**Reply by joep26020** [reply]
ok I got it.

**Reply by joep26020** [reply]
what are your settings

**Reply by Totoqoe** [reply]
So when you open the program up, there should be a drop down where you can select your adapter. If you're connected wirelessly to a network, it will probably be called "Wifi". If you're using etherner, then select the ethernet adapter you use.
After doing so, go to `options` and change the max queue setting to 100000.  Then, go back to the main menu and choose `8kpbs` from the drop down that shows network speeds and then choose `outgoing` for traffic direction. 
Once you have that set up, I'd recommend setting a keybind in options that you can use while joining a lobby, you have to time the keybind to when you have somewhat progressed joining the lobby, itll be around 3 seconds after you press the join button on a lobby.  From there, you should see a void screen where it shows you stuck in the wall, you'll stay there until you turn off the emulator, which you can do once the game starts.

**Reply by joep26020** [reply]
thank you! so how do i set up the keybind though

**Reply by joep26020** [reply]
nevermind I got it, thank you!

**Reply by Totoqoe** [reply]
It's in options too! Also don't expect too much! After all, I only did it to find out how the games netcode could be exploited, which is how I came up with the theories that I made.

**Comment by Unk88** [comment]
if you were to delay the data would this make you not show up on the side bar at all?

**Reply by winstxnhdw** [reply]
No clue. @Totoqoe does it?

**Reply by Totoqoe** [reply]
I haven't tested this with a friend yet, I mostly tested on public lobbies that had the [AC] tag in them.

**Comment by Unk88** [comment]
also not sure how the displayed name position is determined but would it be possible to rapidly shift the position making it harder to kick you or it it locked in place when you join

**Comment by winstxnhdw** [comment]
I missed this when I first looked at this discussion but it's not the `SendNewPlayerValues` that needs to be delayed or blocked, because Anti-Kick already does that. It's probably some other method.

**Reply by Totoqoe** [reply]
I'm moreso wondering what happens if you delay returning false to `SendNewPlayerValues` instead of doing so as soon as ya join?

**Reply by winstxnhdw** [reply]
When anti-kick is enabled, `SendNewPlayerValues` is **never** called.

**Reply by Totoqoe** [reply]
Ohhh, I wonder why it doesn't kick players who get stuck on the wall even though the glitch presumably causes `SendNewPlayerValues` to never be called.

**Reply by winstxnhdw** [reply]
From my experience, I noticed that whenever I get stuck in the wall, it's because my game hit an exception while joining. Usually it's some IndexOutOfBounds error.

**Reply by Totoqoe** [reply]
I'm guessing it would be caused by something on the host side instead of the client?

**Reply by winstxnhdw** [reply]
No idea. But since you are able to replicate it using a network throttler, maybe not?

**Reply by Totoqoe** [reply]
Yeah but typically its temporary when using it, as soon as I turn off the throttle, it causes me to escape the wall and then get kicked if I had anti kick on.  I also can't hear anyone's voices while im still stuck until i fully load in.  
The weird glitch does allow you to hear voices and it caused you to be permanently stuck, while also appearing as nameless or a random player.
